adapter-base-spring/src/main/java/org/eclipse/hono/adapter/spring/AbstractAdapterConfig.java

```diff
@@ -789,8 +790,9 @@ public NotificationReceiver notificationReceiver() {
         if (kafkaConsumerConfig.isConfigured()) {
             return new KafkaBasedNotificationReceiver(vertx(), kafkaConsumerConfig);
         } else {
-            // TODO provide AMQP based notification receiver
-            return new NoOpNotificationReceiver();
+            final ClientConfigProperties notificationConfig = new ClientConfigProperties(downstreamSenderConfig());
+            notificationConfig.setServerRole("Notification");
+            return new ProtonBasedNotificationReceiver(HonoConnection.newConnection(vertx(), notificationConfig, getTracer()));
```

Comment:
Note: I'm using the `hono.messaging` AMQP client configuration here (`downstreamSenderConfig()`). not requiring an extra `hono.notification` configuration block. 
I think this makes the configuration easier and doesn't require a configuration change when upgrading an existing AMQP-messaging based deployment to Hono 1.11.

One might argue that in the Kafka-messaging case there is indeed a separate `hono.kafka.notification` block. But it isn't mandatory and omitting it will work in the usual case that `hono.kafka.commonClientConfig` is set.

If we want to add more flexibility to the AMQP-based notification client configuration, we could define a `hono.notification` config block with `hono.messaging` being used as its default value. @sophokles73 WDYT? Should we add this here?

Reply:
AFAIC it is ok to use the `hono.messaging` configuration in this case. However, I think that we need to make it clear in the configuration guide that the notifications will be received via Kafka by default (if Kafka is configured) and the AMQP messaging network will be used as a fallback only.

Reply:
Yes, I'll update #3022 accordingly.